strings_lab.js

```diff
@@ -0,0 +1,178 @@
+
+let string = "itsy bitsy spider";
+
+let newString = string.slice(0,10)
+console.log(newString)
+
+// ## Bonus Questions
+//
+// 1. Write a code block that adds a "#" to a string if the string has an odd number of characters, otherwise leave it as is.
+// print/log the result
+//
+//     Examples:
+//
+//   ```js
+//   let name = 'taq'  // should log 'taq#'
+//   let name2 ='john' // should log 'john'
+
+let strName = "olive";
+
+if (strName.length % 2) {
+  console.log(strName + "#")
+} else {
+  console.log(strName)
+}
+
+
+
+// 2. Write a code block that prints/logs the 3rd to last characters of a string.
+// If the string is not at least 3 characters long then log an error message.
+// <details>
+//     <summary>
+//         Hint
+//     </summary>
+//     It should work for strings of any length.
+// </details>
+//
+// Examples, for variables:
+// ```js
+// let name1 = 'Taq' // Should log 'Taq'
+// let sentence = 'He is Taq'// Should log 'Taq'
+let sentence= "what do you say to the god of death ";
+let name1 ="go"
+let findVar = sentence.includes(name1)
+
+if ( name1.length < 3){
+    console.log("Invalid character amount, enter word with more than 3 characters")
+} else if (findVar === true){
+    console.log(name1)
+} else {
+    console.log("That word can't be found")
+}
+// 3. Write a code block that prints/logs whether a string `x` starts with a vowel.
+// <details>
+//   <summary>
+//     Extra
+//   </summary>
+//
+//   Make it [case insensitive](https://en.wikipedia.org/wiki/Case_sensitivity)
+//
+// </details>
+let begLetter = "apple";
+let gut = begLetter[0];
+    //console.log(gut)
+ if (gut === ( "a" ) || "e"|| "A" || "E"){
+    console.log(begLetter);
+  } else if (gut === "i" || "o" || "u" || "I" || "O" || "U") {
+      console.log(begLetter)
+   }
+
+
+// 4. Write a code block that given two strings, `str1` and `str2`, prints/logs a new string with `str2` mashed into the middle of `str1`
+//
+// Examples:
+//
+//   - For `str1` = 'help' and `str2` = 'me'. Your program should log `'hemelp'`
+//
+//    - For `str1` = 'hello' and `str2` = 'world'. Your program should log `'helworldlo'`
+let str1 = "inside"
+let str2 = "put"
+let copy1 = str1.slice(0,3)
+let copy2 = str1.slice(3,6)
+console.log(copy1 + str2 + copy2)
+// 5. Write a code block that prints/logs whether a string `x` has at least one vowel in it.
+//
+// Examples:
+//   - For `x` = 'taq'. Your program should log `'taq' has a vowel'`
+//   - For `x` = 'dfghjkl'. Your program should log `'dfghjkl' does not have a vowel'`
+//   - For `x` = '' Your program should log `'' does not have a vowel'`
+//
+// <details>
+//   <summary>
+//     Extra & Hint:
+//   </summary>
+//
+//   - **Hint**: Abstain from using loops. Explore string methods.
+//
+//   - **Extra**: Make it [case insensitive](https://en.wikipedia.org/wiki/Case_sensitivity). So that for `x` = 'TAQ' it should log
+//   `'TAQ' has a vowel`
+// </details>
+let x = "psycho";
+if (x.match(/[aeiouAEIOU]/)) {
```

Comment:
Good use of regex here. Regex is useful thing to learn to make your life easier later on.